components/Ripple/src/MDCRippleTouchController.h

```diff
@@ -46,6 +46,16 @@
  */
 @property(nonatomic, strong, readonly, nonnull) UILongPressGestureRecognizer *gestureRecognizer;
 
+
+/**
+ If set to NO, the ripple gesture will fail and not be initiated if there are other competing
+ gestures that belong to a UIScrollView. This helps the ripple not be invoked if a user wants
+ to scroll but does so while tapping on a view that incorporates a ripple.
+
+ Defaults to YES.
```

Comment:
What's an example of when we'd want this to be YES? Based on the UX shown in the videos I'd assume we'd want `NO` to be the default.

Reply:
The reason i defaulted to YES is because I want to take a cautious approach as I am not sure what gestures could be invoked with their UIScrollView or its subclasses, and if for instance a long press or a tap gesture recognizer is an added gesture, then the ripple will never be invoked as those gesture will cause the ripple gesture to fail. A good example is for instance the dragged behavior in UICollectionViews from iOS 9, which is a gesture that is part of the UICollectionView and will be invoked with a long press.

Therefore I would prefer the client to recognize if this behavior is desirable for them and opt in.

I have also tried to identify if the gesture is a long press or a tap and let the ripple gesture not fail, or only fail when there is a swipe or drag gesture, but I quickly found out that Apple uses a lot of internal gesture classes and I wouldn't want to depend on internal classes especially when they can easily be changed/updated by Apple.

Reply:
I see; is this something where we can be more focused in indicating which gestures should cause the handler to cancel?

It sounds like we might be looking at too many gesture recognizers?

Reply:
Yeah, the system gestures just from UICollectionView I've been able to gather are called:
`_UISystemGestureGateGestureRecognizer`, `UIScreenEdgePanGestureRecognizer`, `UIScrollViewDelayedTouchesBeganGestureRecognizer`, `UIScrollViewPanGestureRecognizer`, `UILongPressGestureRecognizer`, `UIScreenEdgePanGestureRecognizer`, `UIScrollViewPanGestureRecognizer`.

As you can see most of them are internal, and I think it would be unsafe to call them out in the method to try and be more focused. I could specifically choose known gesture recognizers like Tap and LongPress as gestures that shouldn't fail the ripple to help improve the behavior though if you think that makes sense.

Reply:
> I could specifically choose known gesture recognizers like Tap and LongPress as gestures that shouldn't fail the ripple to help improve the behavior though if you think that makes sense.

If this has a chance of letting us make the desired behavior be the default then I think it's worth the exploration. This feels like a thing we could possibly iterate on independent of the ability to opt in to this behavior (which is my understanding of what this new property does).

Reply:
Filed https://github.com/material-components/material-components-ios/issues/7314 to track